sdk/core/System.ClientModel/src/Options/LoggingOptions.cs

```diff
@@ -0,0 +1,148 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.Abstractions;
+
+namespace System.ClientModel.Primitives;
+
+/// <summary>
+/// Exposes client options related to logging.
+/// </summary>
+public class LoggingOptions
+{
+    private bool _frozen;
+
+    private const int DefaultLoggedContentSizeLimit = 4 * 1024;
+    private const bool DefaultIsLoggingContentEnabled = false;
+    private const bool DefaultCombineLogs = true;
+
+    private static readonly string[] s_defaultAllowedHeaderNames =
+        new[] {
+            "traceparent",
+            "Accept",
+            "Cache-Control",
+            "Connection",
+            "Content-Length",
+            "Content-Type",
+            "Date",
+            "ETag",
+            "Expires",
+            "If-Match",
+            "If-Modified-Since",
+            "If-None-Match",
+            "If-Unmodified-Since",
+            "Last-Modified",
+            "Pragma",
+            "Request-Id",
+            "Retry-After",
+            "Server",
+            "Transfer-Encoding",
+            "User-Agent",
+            "WWW-Authenticate" };
+    private static readonly string[] s_defaultAllowedQueryParameters = new[] { "api-version" };
+
+    private int _loggedContentSizeLimit = DefaultLoggedContentSizeLimit;
+    private bool _isLoggingContentEnabled = DefaultIsLoggingContentEnabled;
+    private IList<string> _allowedHeaderNames = new List<string>(s_defaultAllowedHeaderNames);
+    private IList<string> _allowedQueryParameters = new List<string>(s_defaultAllowedQueryParameters);
+    private ILoggerFactory _loggerFactory = NullLoggerFactory.Instance;
+
+    /// <summary>
+    /// Gets or sets the implementation of <see cref="ILoggerFactory"/> to use to
+    /// create <see cref="ILogger"/> instances for logging.
+    /// </summary>
+    /// <remarks>If an ILoggerFactory is not provided, logs will be written to Event Source
+    /// instead. If an ILoggerFactory is provided, logs will be written to ILogger only and not
+    /// Event Source.</remarks>
+    /// <value>Defaults to <see cref="NullLoggerFactory"/>.</value>
+    public ILoggerFactory LoggerFactory
```

Comment:
A downside with this approach in general is that if someone passes in an ILogger and also has a listener, it may be confusing why they don't get any events written to the event source. I could imagine frustration if someone is new to event source and thinks they are not seeing events because they are doing something wrong.

Although, this is probably somewhat unlikely because I don't think it's common for anyone to need multiple sets of the same logs.